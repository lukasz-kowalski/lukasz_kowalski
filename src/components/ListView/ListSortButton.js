import React from 'react';

class ListSortButton extends React.Component {
  handleClick = () => {
    this.props.handleSortChange(this.props.name.toLowerCase());
  }

  render() {
    const { name } = this.props;
    return (
      <th scope="col">
      {name !== 'Actions' ?
        <button 
          className="th-button" 
          onClick={this.handleClick}
        >
          {name}
        </button> :
        name
      }
      </th>
    );
  }
}

export default ListSortButton;
