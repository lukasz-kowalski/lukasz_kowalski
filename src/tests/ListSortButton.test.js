import React from 'react';
import { shallow } from 'enzyme';
import ListSortButton from '../components/ListView/ListSortButton';

let wrapper;

const handleSortChange = jest.fn();

beforeEach(() => {
  wrapper = shallow(
  <ListSortButton
    name="Species"
    handleSortChange={handleSortChange}
  />);
})

test('should successful render ListSortButton', () => {
  expect(wrapper).toMatchSnapshot();
})

test('should call handleSortChange when button click', () => {
  wrapper.find('.th-button').simulate('click');
  expect(handleSortChange).toBeCalled();
})

test('should call handleSortChange with name arg when button click', () => {
  const name = 'Species'.toLowerCase();
  wrapper.find('.th-button').simulate('click');
  expect(handleSortChange).toBeCalledWith(name);
})
