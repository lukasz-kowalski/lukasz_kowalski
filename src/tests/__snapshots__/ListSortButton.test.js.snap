// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successful render ListSortButton 1`] = `
<th
  scope="col"
>
  <button
    className="th-button"
    onClick={[Function]}
  >
    Species
  </button>
</th>
`;
